### PR TITLE
chore(build): Add base image Dockerfile for building kepler with libbpf

### PR DESCRIPTION
**Pull request by vimalk78**
- Adds Dockerfile.base which installs everything required to build kepler code base
  - Updated github workflow to build and push the base image

### CODE REVIEW

**Comment by vimalk78** [comment]
cc: @rootfs

**Comment by vimalk78** [comment]
@rootfs `base` job seems to run when push happens to `base` branch. so should this PR be targeted for `base` branch?

File: Dockerfile.base [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+RUN yumdownloader --source libbpf
+RUN rpm2cpio libbpf-1.2.0-1.el9.src.rpm | cpio -iv
+RUN tar xf ./linux-*el9.tar.xz
+WORKDIR /libbpf-source/linux-5.14.0-333.el9/tools/lib/bpf
+RUN make install_headers
+RUN prefix=/usr BUILD_STATIC_ONLY=y make install
+WORKDIR /libbpf-source/linux-5.14.0-333.el9/tools/bpf
+RUN make bpftool
+
+RUN yum -y install clang
```
Comment (rootfs) [inline Dockerfile.base:28]: can you move build tools to the builder image? the base image only has the runtime dependency for kepler.

Comment (vimalk78) [reply Dockerfile.base:28]: @rootfs the `Dockerfile.base` is the builder image. may be it should be renamed as such?

Comment (rootfs) [reply Dockerfile.base:28]: we have 3 dockerfiles: base, builder, and kepler

The build tools are in the builder image, e.g. this one https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile.libbpf.builder

Comment (vimalk78) [reply Dockerfile.base:28]: but in my refactoring there are only two Dockerfiles. `Dockerfile.base` which has all the things needed to build kepler. and `Dockerfile` which has a stage to build kapler and next stage copies everything needed to run kepler.
https://github.com/sustainable-computing-io/kepler/pull/1068

File: .github/workflows/image_base.yml [hunk 2]
```diff
@@ -53,5 +53,13 @@ jobs:
...
           push: true
           tags: quay.io/sustainable_computing_io/kepler_base:latest-libbpf-amd64
+      - name: Build and push a base image for building Kepler with libbpf
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./Dockerfile.base
+          platforms: linux/amd64
+          push: true
+          tags: quay.io/sustainable_computing_io/kepler_base:latest-libbpf-amd64-simplified
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply .github/workflows/image_base.yml:63]: i can add the same tag, but kept it separate to not overwrite old images.

File: Dockerfile.base (hunk 1 above)
Comment (rootfs) [reply Dockerfile.base:28]: Got it, so the new base is just ubi9 mimal.

**Review by rootfs — APPROVED** [review]

File: .github/workflows/image_base.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rootfs) [reply .github/workflows/image_base.yml:63]: let's add a new image first and test build the kepler image. If everything works, we can set the new tag and remove the existing build dockerfiles.

**Comment by vimalk78** [comment]
will this merge trigger the action to build base image?

**Pull request merged**